Elevel/Application/Features/TestCommands/GetAllTestsQuery.cs

```diff
@@ -0,0 +1,98 @@
+﻿using AutoMapper;
+using AutoMapper.QueryableExtensions;
+using Elevel.Application.Interfaces;
+using Elevel.Application.Pagination;
+using Elevel.Domain;
+using Elevel.Domain.Enums;
+using Elevel.Domain.Models;
+using MediatR;
+using Microsoft.AspNetCore.Identity;
+using Microsoft.EntityFrameworkCore;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Linq.Expressions;
+using System.Threading;
+using System.Threading.Tasks;
+
+
+namespace Elevel.Application.Features.TestCommands
+{
+    public class GetAllTestsQuery
+    {
+        public class Request : PagedQueryBase, IRequest<Response>
+        {
+            public Level? Level { get; set; }
+            public DateTimeOffset? TestPassingDate { get; set; }
+        }
+
+        public class Handler : IRequestHandler<Request, Response>
+        {
+            private readonly IApplicationDbContext _context;
+            private readonly IMapper _mapper;
+            public Handler(IApplicationDbContext context, IMapper mapper)
+            {
+                _context = context;
+                _mapper = mapper;
+            }
+            public async Task<Response> Handle(Request request, CancellationToken cancelationtoken)
+            {
+                var tests = _context.Tests.AsNoTracking();
+
+                if (request.Level!=null)
```

Comment:
The better way to check null in nullable type is HasValue property

Reply:
it's clearer but both ways the same as I remember

Reply:
https://stackoverflow.com/questions/676078/what-is-the-difference-between-nullablet-hasvalue-or-nullablet-null